src/compiler/scanner.rs

```diff
@@ -73,13 +126,43 @@ impl Scanner {
                     return self.set_token(SyntaxKind::SemicolonToken);
                 }
                 _ch => {
-                    unimplemented!();
-                    // let identifier_kind = self.scan_identifier(ch);
+                    let identifier_kind = self.scan_identifier(ch);
+                    if let Some(identifier_kind) = identifier_kind {
+                        return self.set_token(identifier_kind);
+                    }
+                    unimplemented!()
                 }
             }
         }
     }
 
+    fn scan_identifier(&mut self, start_character: char) -> Option<SyntaxKind> {
+        let mut ch = start_character;
+        if is_identifier_start(ch) {
+            self.set_pos(self.pos() + char_size(ch));
+            loop {
+                if !(self.pos() < self.end()) {
+                    break;
+                }
+                ch = code_point_at(self.text(), self.pos());
+                if !is_identifier_part(ch) {
+                    break;
+                }
+                self.set_pos(self.pos() + char_size(ch));
+            }
+            self.set_token_value(
+                &self
+                    .text()
+                    .chars()
+                    .skip(self.token_pos())
+                    .take(self.pos() - self.token_pos())
+                    .collect::<String>(),
```

Comment:
Is the `<String>` here required or for clarity? (I'm surprised it couldn't infer it, since it's being used directly as a param to `set_token_value`.)

Reply:
If I omit the `<String>` it fails with ``value of type `str` cannot be built from `std::iter::Iterator<Item=char>` `` - I think the inference issue would be that `.set_token_value()` takes an `&str`, so without some "`String` deref's into `str`" "extra intelligence" it would want to assume that this expression that's being `&`'ed should be typed as `str`, not `String`